runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/task/TaskExecutor.java

```diff
@@ -310,21 +320,59 @@ private void doExecute() {
     }
   }
 
+  public void sendDynOptData(final Object element) {
+    Map<Object, Long> dynOptData = (Map<Object, Long>) element;
+    final List<ControlMessage.PartitionSizeEntry> partitionSizeEntries = new ArrayList<>();
+    dynOptData.forEach((key, size) ->
+      partitionSizeEntries.add(
+        ControlMessage.PartitionSizeEntry.newBuilder()
+          .setKey(key == null ? "" : String.valueOf(key))
```

Comment:
What does the null key and the empty string mean? Use a static variable for the empty string?

Reply:
There were cases when keys are `null` which lead to exception. I'll add `NULL_KEY` static variable string for it. Thanks!